### PR TITLE
Make the metadata loader use the appropriate Target structure

### LINKED ISSUE #19907: Compiler plugins not found with target spec
When using --target <json-file> rustc will fail to find compiler plugins.

I suspect it might not look for plugins for the host anymore.

### PR DESCRIPTION
**Pull request by Zoxc**
Fixes #19907

### CODE REVIEW

File: src/librustc/metadata/loader.rs [hunk 1]
```diff
@@ -248,6 +249,7 @@ pub struct Context<'a> {
     pub ident: &'a str,
     pub crate_name: &'a str,
     pub hash: Option<&'a Svh>,
+    pub target: &'a Target,
```
Comment (huonw) [inline src/librustc/metadata/loader.rs]: Could you add a comment about how this differs to `sess.target`?

File: src/librustc/session/mod.rs [hunk 2]
```diff
@@ -243,6 +246,12 @@ pub fn build_session_(sopts: config::Options,
                       local_crate_source_file: Option<Path>,
                       span_diagnostic: diagnostic::SpanHandler)
                       -> Session {
+    let host = match Target::search(config::host_triple()) {
+        Ok(t) => t,
+        Err(e) => {
+            span_diagnostic.handler().fatal((format!("Error loading host specification: {}", e)).as_slice());
```
Comment (huonw) [inline src/librustc/session/mod.rs]: This line is too long (over 100 chars) and is failing travis.

File: src/librustc/metadata/loader.rs (hunk 1 above)
Comment (Zoxc) [inline src/librustc/metadata/loader.rs]: The `creader.rs` changes sets it to either &self.sess.target.target or &self.sess.host depending on whether it looks for target or host libraries.

Comment (huonw) [inline src/librustc/metadata/loader.rs]: I was meaning add one to the source. :)

**Pull request merged**